test/integration/routes/member.js

```diff
@@ -0,0 +1,112 @@
+const { get, set } = require('@0ti.me/tiny-pfp');
+const {
+  HTTP_METHODS: { DELETE, GET, PUT },
+  JSON_SELECTORS: { POOL },
+} = require('../../../src/lib/constants');
+const { OK } = require('http-status-codes');
+const supertestCreator = require('../../lib/supertest');
+
+const { _, bluebird, d, nextInt, uuid } = deps;
+
+const me = __filename;
+
+d(me, () => {
+  let config = null;
+  let context = null;
+  let data = null;
+  let supertest = null;
+
+  beforeEach(() => {
+    data = `data-${uuid()}`;
+
+    context = {};
+
+    config = _.clone(require('config'));
+
+    set(context, 'config', config);
+
+    const promises = {};
+
+    promises.supertest = supertestCreator.initializeSupertest(context);
+
+    return bluebird.props(promises).then((resolutions) => {
+      supertest = resolutions.supertest;
+    });
+  });
+
+  describe(GET, () => {
+    beforeEach(() => {
+      get(context, POOL).bindQueryAndResponse(
+        'SELECT * FROM "Member" LIMIT \'10\'',
+        {
+          rows: data,
+        },
+      );
+    });
+
+    it('should respond with OK and the result of the query', () =>
+      supertest.get('/api/member').expect(OK, { data }));
+  });
+
+  describe(DELETE, () => {
+    let id = null;
+
+    beforeEach(() => {
+      id = nextInt();
+
+      get(context, POOL).bindQueryAndResponse(
+        `DELETE FROM "Member" WHERE id = $1;`,
+        [id],
+        {
+          rows: data,
+        },
+      );
+    });
+
+    it('should respond with OK and the empty data rows result', () =>
+      supertest.delete('/api/member').send({ id }).expect(OK, { data }));
+  });
+
+  describe(PUT, () => {
+    let name = null;
+
+    beforeEach(() => {
+      name = `name-${uuid()}`;
+    });
+
+    describe('create', () => {
+      beforeEach(() => {
+        get(
+          context,
+          POOL,
+        ).bindQueryAndResponse(
+          `INSERT INTO "Member" ( name ) VALUES ( $1 );`,
+          [name],
+          { rows: data },
+        );
+      });
+
+      it('should respond with OK and the empty data rows result', () =>
+        supertest.put('/api/member').send({ name }).expect(OK, { data }));
+    });
+
+    describe('update', () => {
+      let id = null;
+
+      beforeEach(() => {
+        id = nextInt();
+        get(
+          context,
+          POOL,
+        ).bindQueryAndResponse(
+          `UPDATE "Member" SET name = $1 WHERE id = $2;`,
+          [name, id],
```

Comment:
Where's `name` come from?

Reply:
Line 70, the outer describe sets up the `name` value.

For better or worse, in these test files, I try to minimize instantiation of variables just because I don't want the test files to be even longer than they have to be.

Since multiple tests use `name`, I moved it up top.

I'm not sure if you're familiar with beforeEach, but in each describe block, the beforeEach callbacks are called from shallowest to deepest in the order they're created.

For example:
```
describe('bla', () => {
  beforeEach(() => 1); // 1
  beforeEach(() => 2); // 2
  describe('fred', () => {
    beforeEach(() => 3); // 3
  });
});
```

Those are called in the order [1, 2, 3].

Reply:
Ahh, I see it now, thanks :)